lambdas/indexer/Makefile

```
@@ -15,6 +15,11 @@ wheels: check_env
 # TODO: Rename target: https://github.com/DataBiosphere/azul/issues/1654
 deploy: check_branch check_python check_aws config wheels
 	python -m azul.changelog vendor
+	# Even though Python >=3.7 ensures that .pyc files are generated
+	# consistently, it's possible that not all .pyc files will be present when
+	# lambdas are deployed. Running compileall ensures all .pyc files are always
+	# included.
```

```suggestion
	# Ensure a .pyc file is present for every .py file
```

The comment about deterministic pyc's is confusing and bears no relevance in this context.

How fast is `compileall` and is it efficient in skipping the generation of pyc's that already exist?

---

After clearing all compiled files:
```
(.venv) jesse@vader ~/gi/azul (issues/jessebrennan/1673-stable-deploy)$ time python -m compileall lambdas/service/vendor/ lambdas/service/vendor/azul
Listing 'lambdas/service/vendor/'...
Compiling 'lambdas/service/vendor/azul_changes.py'...
Listing 'lambdas/service/vendor/jsonobject'...
Compiling 'lambdas/service/vendor/jsonobject/__init__.py'...
Compiling 'lambdas/service/vendor/jsonobject/exceptions.py'...
Listing 'lambdas/service/vendor/jsonobject-0.9.9.dist-info'...
Listing 'lambdas/service/vendor/static'...
Listing 'lambdas/service/vendor/azul'...
Compiling 'lambdas/service/vendor/azul/__init__.py'...
Compiling 'lambdas/service/vendor/azul/aws_service_model.py'...
Compiling 'lambdas/service/vendor/azul/azulclient.py'...
Compiling 'lambdas/service/vendor/azul/chalice.py'...
Compiling 'lambdas/service/vendor/azul/changelog.py'...
Compiling 'lambdas/service/vendor/azul/collections.py'...
Compiling 'lambdas/service/vendor/azul/deployment.py'...
Compiling 'lambdas/service/vendor/azul/doctests.py'...
Compiling 'lambdas/service/vendor/azul/drs.py'...
Compiling 'lambdas/service/vendor/azul/dss.py'...
Compiling 'lambdas/service/vendor/azul/es.py'...
Compiling 'lambdas/service/vendor/azul/files.py'...
Compiling 'lambdas/service/vendor/azul/health.py'...
Compiling 'lambdas/service/vendor/azul/hmac.py'...
Listing 'lambdas/service/vendor/azul/indexer'...
Compiling 'lambdas/service/vendor/azul/indexer/__init__.py'...
Compiling 'lambdas/service/vendor/azul/indexer/policy.py'...
Compiling 'lambdas/service/vendor/azul/json.py'...
Compiling 'lambdas/service/vendor/azul/json_freeze.py'...
Compiling 'lambdas/service/vendor/azul/lambda_layer.py'...
Compiling 'lambdas/service/vendor/azul/lambdas.py'...
Compiling 'lambdas/service/vendor/azul/logging.py'...
Compiling 'lambdas/service/vendor/azul/modules.py'...
Compiling 'lambdas/service/vendor/azul/objects.py'...
Listing 'lambdas/service/vendor/azul/openapi'...
Compiling 'lambdas/service/vendor/azul/openapi/__init__.py'...
Compiling 'lambdas/service/vendor/azul/openapi/params.py'...
Compiling 'lambdas/service/vendor/azul/openapi/responses.py'...
Compiling 'lambdas/service/vendor/azul/openapi/schema.py'...
Compiling 'lambdas/service/vendor/azul/plugin.py'...
Listing 'lambdas/service/vendor/azul/plugins'...
Listing 'lambdas/service/vendor/azul/plugins/metadata'...
Listing 'lambdas/service/vendor/azul/plugins/metadata/hca'...
Compiling 'lambdas/service/vendor/azul/portal_service.py'...
Listing 'lambdas/service/vendor/azul/project'...
Compiling 'lambdas/service/vendor/azul/project/__init__.py'...
Listing 'lambdas/service/vendor/azul/project/hca'...
Compiling 'lambdas/service/vendor/azul/project/hca/__init__.py'...
Compiling 'lambdas/service/vendor/azul/project/hca/indexer.py'...
Compiling 'lambdas/service/vendor/azul/project/hca/metadata_generator.py'...
Compiling 'lambdas/service/vendor/azul/project/hca/transformers.py'...
Compiling 'lambdas/service/vendor/azul/queues.py'...
Compiling 'lambdas/service/vendor/azul/requests.py'...
Listing 'lambdas/service/vendor/azul/security'...
Compiling 'lambdas/service/vendor/azul/security/__init__.py'...
Compiling 'lambdas/service/vendor/azul/security/authenticator.py'...
Listing 'lambdas/service/vendor/azul/service'...
Compiling 'lambdas/service/vendor/azul/service/__init__.py'...
Compiling 'lambdas/service/vendor/azul/service/async_manifest_service.py'...
Compiling 'lambdas/service/vendor/azul/service/buffer.py'...
Compiling 'lambdas/service/vendor/azul/service/cart_export_job_manager.py'...
Compiling 'lambdas/service/vendor/azul/service/cart_export_service.py'...
Compiling 'lambdas/service/vendor/azul/service/cart_item_manager.py'...
Compiling 'lambdas/service/vendor/azul/service/collection_data_access.py'...
Compiling 'lambdas/service/vendor/azul/service/dynamo_data_access.py'...
Compiling 'lambdas/service/vendor/azul/service/elasticsearch_service.py'...
Compiling 'lambdas/service/vendor/azul/service/hca_response_v5.py'...
Compiling 'lambdas/service/vendor/azul/service/manifest_service.py'...
Compiling 'lambdas/service/vendor/azul/service/policy.py'...
Compiling 'lambdas/service/vendor/azul/service/repository_service.py'...
Listing 'lambdas/service/vendor/azul/service/responseobjects'...
Listing 'lambdas/service/vendor/azul/service/service_config'...
Compiling 'lambdas/service/vendor/azul/service/step_function_helper.py'...
Compiling 'lambdas/service/vendor/azul/service/storage_service.py'...
Compiling 'lambdas/service/vendor/azul/service/user_service.py'...
Compiling 'lambdas/service/vendor/azul/service/utilities.py'...
Compiling 'lambdas/service/vendor/azul/strings.py'...
Compiling 'lambdas/service/vendor/azul/subscription.py'...
Compiling 'lambdas/service/vendor/azul/template.py'...
Compiling 'lambdas/service/vendor/azul/threads.py'...
Compiling 'lambdas/service/vendor/azul/time.py'...
Compiling 'lambdas/service/vendor/azul/transformer.py'...
Compiling 'lambdas/service/vendor/azul/types.py'...
Listing 'lambdas/service/vendor/azul/vendored'...
Compiling 'lambdas/service/vendor/azul/vendored/__init__.py'...
Listing 'lambdas/service/vendor/azul/vendored/frozendict'...
Compiling 'lambdas/service/vendor/azul/vendored/frozendict/__init__.py'...
Compiling 'lambdas/service/vendor/azul/version_service.py'...

real    0m0.109s
user    0m0.108s
sys     0m0.001s
```
Then, without clearing the caches I compiled for the indexer
```
Listing 'lambdas/indexer/vendor/'...
Compiling 'lambdas/indexer/vendor/azul_changes.py'...
Listing 'lambdas/indexer/vendor/jsonobject'...
Compiling 'lambdas/indexer/vendor/jsonobject/__init__.py'...
Compiling 'lambdas/indexer/vendor/jsonobject/exceptions.py'...
Listing 'lambdas/indexer/vendor/jsonobject-0.9.9.dist-info'...
Listing 'lambdas/indexer/vendor/azul'...
Listing 'lambdas/indexer/vendor/azul/indexer'...
Listing 'lambdas/indexer/vendor/azul/openapi'...
Listing 'lambdas/indexer/vendor/azul/plugins'...
Listing 'lambdas/indexer/vendor/azul/plugins/metadata'...
Listing 'lambdas/indexer/vendor/azul/plugins/metadata/hca'...
Listing 'lambdas/indexer/vendor/azul/project'...
Listing 'lambdas/indexer/vendor/azul/project/hca'...
Listing 'lambdas/indexer/vendor/azul/security'...
Listing 'lambdas/indexer/vendor/azul/service'...
Listing 'lambdas/indexer/vendor/azul/service/responseobjects'...
Listing 'lambdas/indexer/vendor/azul/service/service_config'...
Listing 'lambdas/indexer/vendor/azul/vendored'...
Listing 'lambdas/indexer/vendor/azul/vendored/frozendict'...

real    0m0.074s
user    0m0.062s
sys     0m0.012s
```
>  is it efficient in skipping the generation of pyc's that already exist?
It only compiles files if the timestamp is older then the source file, so yes.

I'm considering adding `-q` for more quiet output. What do you think?

---

Yes